src/canvas/SelectableCanvas.ts

```diff
@@ -680,31 +684,46 @@ export class SelectableCanvas<
           target.cacheTranslationY + normalizedPointer.y * target.zoomY,
           0
         );
+      // transform tolerance according to vpt
+      // TODO: use sendVectorToPlane
+      const tolerance = new Point()
+        .scalarAdd(this.targetFindTolerance)
+        .transform(invertTransform(this.viewportTransform), true);
+      const size = tolerance.scalarMultiply(2).max(new Point(1, 1));
+      // performance optimization:
+      // we draw the hit area to the dedicated canvas instead of using `getImageData` on the target's cache canvas
+      // since `size` is transformed according to vpt the image is drawn as if transformed as well so `targetFindTolerance` can be used as the tolerance value
+      ctx.drawImage(
+        target._cacheCanvas,
+        Math.floor(targetRelativeX - tolerance.x),
+        Math.floor(targetRelativeY - tolerance.y),
+        Math.ceil(size.x),
+        Math.ceil(size.y),
+        0,
+        0,
+        Math.ceil(size.x * retina),
+        Math.ceil(size.y * retina)
+      );
 
       return isTransparent(
-        target._cacheContext,
-        Math.round(targetRelativeX),
-        Math.round(targetRelativeY),
+        ctx,
+        tolerance.x,
+        tolerance.y,
         this.targetFindTolerance
       );
     }
 
-    const ctx = this.contextCache,
-      originalColor = target.selectionBackgroundColor,
-      v = this.viewportTransform;
-
-    target.selectionBackgroundColor = '';
-
-    this.clearContext(ctx);
```

Comment:
who is clearing the context now between different checks? and where the `selectionBackgroundColor` handling code went?

Reply:
oops.

Reply:
I will restore selection background... 
The context must be scaled because we set dimensions of canvas as well and never re init retina, right?

Reply:
not sure. i m building a little demo based on this concept so we can talk with examples.
A fun diversion from types/tests

Reply:
> A fun diversion from types/tests

cool

Since you merged the sandbox pr just run
`npm run sandbox start -- -t next [path/to/folder or nothing]`
if you provide a path the app will build there, if not it will build on the template. Which is fine as well.

Reply:
This is the deployed version
https://codesandbox.io/p/sandbox/fabric-nextjs-sandbox-forked-k53evn?welcome=true&file=%2FREADME.md

Can you look at the sandbox github app and see it can operate in the repo? It should post links for us to reproduce with ease using the templates I built that are now merged into master

Reply:
The app I posted is nothing really, just objects on canvas using the next template

Reply:
i will look into this sandbox as soon as i can.
The goal is that for each PR there is a sandbox with the branch built?

Reply:
correct